### ISSUE TITLE
Nuclear Norm Subgradient 

### ISSUE THREAD

**Issue by josh0tt** [cvxpy/cvxpy#2364]
**Describe the bug**
The current implementation of the subgradient computation for the nuclear norm in CVXPY seems to only handle square matrices correctly due to its use of the `np.linalg.svd(values[0])` method without specifying `full_matrices=False` [here](https://github.com/cvxpy/cvxpy/blob/master/cvxpy/atoms/norm_nuc.py#L50)

**To Reproduce**

```
import cvxpy as cp
import numpy as np

n = 10
m = 20 
z = cp.Variable((n, m))
z.value = np.random.randn(n, m)
objective = cp.Minimize(cp.norm(z, "nuc"))

arg_values = []
for arg in objective.expr.args:
    arg_values.append(arg.value)
    
objective.expr.grad(arg_values)
```

**Expected behavior**
The subgradient of the nuclear norm should be able to handle matrices of size nxm which is also consistent with cvxpy atomic functions table. 

**Output**

```
File ~cvxpy/atoms/atom.py:402, in Atom.grad(self)
    399         arg_values.append(arg.value)
    401 # A list of gradients w.r.t. arguments
--> 402 grad_self = self._grad(arg_values)
    403 # The Chain rule.
    404 result = {}

File ~cvxpy/atoms/norm_nuc.py:51, in normNuc._grad(self, values)
     49 # Grad UV^T
     50 U, _, V = np.linalg.svd(values[0])
---> 51 D = U.dot(V)
     52 return [sp.csc_matrix(D.ravel(order='F')).T]

ValueError: shapes (10,10) and (20,20) not aligned: 10 (dim 1) != 20 (dim 0)
```

**Version**
 - OS: macOS 14.1
 - CVXPY Version: 1.4.2

**Additional context**
I believe the solution is to set `U, _, V = np.linalg.svd(values[0], full_matrices=False)`

**Comment by chofchof** [comment]
@josh0tt You're right. Without `full_matrices=False`, the 51th line `D = U.dot(V)` in `cvxpy/atoms/norm_nuc.py` did not work.

Anyway, what do you want with the last line `objective.expr.grad(arg_values)` of your example?

**Comment by josh0tt** [comment]
I was originally trying to use the DCCP package [here](https://github.com/cvxgrp/dccp). In order to do that, we need to linearize the objective. For the nuclear norm, this involves using the subgradient. So not exactly related to what I showed in the example, that was just the minimum working example I came up with and the last block came from `atom.py` [here](https://github.com/cvxpy/cvxpy/blob/083d7854e036cfcccd184f50b237e74c7dbfea82/cvxpy/atoms/atom.py#L394C1-L399C45).
